config/routes.rb

```diff
@@ -1,6 +1,13 @@
 Rails.application.routes.draw do
+  get '/login', to: 'sessions#new'
+  post '/login', to: 'sessions#create'
+  delete '/logout', to: 'sessions#destroy'
+  get '/signup', to: 'users#new'  
+  post 'products/add_to_cart/:id', to: 'products#add_to_cart', as: 'add_to_cart'
+  delete 'products/remove_from_cart/:id', to: 'products#remove_from_cart', as: 'remove_from_cart'
```

Comment:
you could move these 2 into the `resources :products` and have something like this:
```ruby
resources :products, only: %i[index new create update etc], param: :product_id do # the values in `only` are optional, add only the methods you need
  post :add_to_cart, on: :collection # will look like POST /products/add_to_cart/:product_id
  delete : remove_from_cart, on: :collection # will look like DELETE /products/remove_from_cart/:product_id
end
```

Reply:
I successfully converted add_to_cart & remove_from_cart, but I don't know how to convert the commented routes.
`  resources :products, only: %i[index cart] do
    post :add_to_cart, on: :member
    delete :remove_from_cart, on: :member
  end
  # get '/cart', to: 'products#cart', as: 'cart'
  # get '/dashboard', to: 'products#index', as: 'dashboard'`

Reply:
`  resources :products, only: %i[index cart] do
    post :add_to_cart, on: :member
    delete :remove_from_cart, on: :member
  end`

Reply:
`resources :products, only: %i[index cart] do
  post :add_to_cart`

Reply:
`  resources :products, only: %i[index cart] do
    post :add_to_cart, on: :member
    delete :remove_from_cart, on: :member
  end
  # get '/cart', to: 'products#cart', as: 'cart'
  # get '/dashboard', to: 'products#index', as: 'dashboard'
  `

Reply:
Moving them into the `resources :products` will makes the url be `products/method`. If you don't want that, it's ok to let them outside the resource